job-manager/jobstore-api/src/test/java/uk/gov/moj/cpp/jobstore/api/task/ExecutionInfoTest.java

```diff
@@ -84,4 +85,48 @@ public void shouldSetTheExecutionStatus() {
 
         assertThat(copiedExecutionInfo.getExecutionStatus(), is(ExecutionStatus.STARTED));
     }
+
+    @Test
+    public void shouldBuildWhenShouldRetryIsTrueAndExhaustTaskDetailsSupplied() {
+        ExecutionInfo copiedExecutionInfo = ExecutionInfo.executionInfo().withShouldRetry(true)
+                .withNextTask("last-task")
+                .withNextTaskStartTime(ZonedDateTime.now())
+                .withJobData(jobData)
+                .build();
+
+        assertThat(copiedExecutionInfo.getJobData(), is(jobData));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenRetryExhaustTaskJobDataIsNullAndShouldRetryIsTrue() {
+        var e = assertThrows(InvalidRetryExecutionInfoException.class, () -> ExecutionInfo.executionInfo().withShouldRetry(true)
```

Comment:
These tests would be easier to read in you don't use 'var' but instead use 'final InvalidRetryExecutionInfoException ...'